static/src/javascripts/projects/common/modules/commercial/close-disabled-slots.js

```diff
@@ -0,0 +1,58 @@
+define([
+    'bonzo',
+    'qwery',
+    'Promise',
+    'common/utils/$css',
+    'common/utils/fastdom-promise',
+    'common/modules/commercial/commercial-features'
+], function (
+    bonzo,
+    qwery,
+    Promise,
+    $css,
+    fastdom,
+    commercialFeatures
+) {
+    var adSlotSelector = '.js-ad-slot';
+
+    return {
+        init: init
```

Comment:
This was moved out of the `dfp-api` module to fix a race condition. I think this issue affects your [PR](https://github.com/guardian/frontend/pull/13289/files#diff-844a43280e1947d8dd7cc7a480fd0551R334) too, in `queueAdvert`. If we initialise `dfp-api`, and begin defining slots, then we must wait for all visually hidden slots to be removed from the dom before we use the gpt library. For most situations, the removed slots do not interfere with the new defined slots. However, this change introduces `top-above-nav` to mobile for fabric. So now the removed slots contains a `top-above-nav` (the old one rendered by the server as a top banner), and the new slots also contains a `top-above-nav` (the one on mobile fronts under the first container).

So the `close-disabled-slots` module has the same functionality as before, just wrapped in a promise. This means we can relocate this system further up commercial boot's `then` chain.